### PR TITLE
test(e2e): test TypeScript 5.2, beta, latest

### PR DESCRIPTION
**Pull request by Jolg42**
Random 10 min thing I did and added as "tech debt" work.

beta: https://devblogs.microsoft.com/typescript/announcing-typescript-5-3-beta/

Added
- 5.2.x, currently `5.2.2`
- beta, currently `5.3.0-beta`
- latest, currently `5.2.2` (but will switch to 5.3.x once 5.3 is released)

See https://www.npmjs.com/package/typescript?activeTab=versions
Screenshot 2023-11-10 at 10 34 23

### CODE REVIEW

**Comment by Jolg42** [comment]
Note: tests are passing (see [here](https://github.com/prisma/prisma/actions/runs/6823086245/job/18556336130?pr=21884#step:5:279))

File: packages/client/tests/e2e/ts-version/5.1/package.json [hunk 1]
```diff
@@ -10,6 +10,6 @@
     "@types/jest": "29.5.1",
     "@types/node": "16.18.11",
     "prisma": "/tmp/prisma-0.0.0.tgz",
-    "typescript": "~5.1.6"
+    "typescript": "~5.2.2"
```
Comment (aqrln) [inline packages/client/tests/e2e/ts-version/5.1/package.json:13]: `5.1` and `5.2` are mixed up, `5.1` has `"typescript": "~5.2.2"` and `5.2` has `"typescript": "~5.1.6"`

File: packages/client/tests/e2e/ts-version/5.2/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Readme
+
+This is testing TypeScript version 5.1.x
```
Comment (aqrln) [inline packages/client/tests/e2e/ts-version/5.2/README.md:3]:
```suggestion
This is testing TypeScript version 5.2.x
```

File: packages/client/tests/e2e/ts-version/5.1/package.json (hunk 1 above)
Comment (Jolg42) [reply packages/client/tests/e2e/ts-version/5.1/package.json:13]: Nice 🙈 
Thanks!

Comment (Jolg42) [inline packages/client/tests/e2e/ts-version/5.1/package.json:13]:
```suggestion
    "typescript": "~5.1.6"
```

File: packages/client/tests/e2e/ts-version/5.2/package.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  "main": "index.js",
+  "scripts": {},
+  "dependencies": {
+    "@prisma/client": "/tmp/prisma-client-0.0.0.tgz"
+  },
+  "devDependencies": {
+    "@types/jest": "29.5.1",
+    "@types/node": "16.18.11",
+    "prisma": "/tmp/prisma-0.0.0.tgz",
+    "typescript": "~5.1.6"
```
Comment (Jolg42) [inline packages/client/tests/e2e/ts-version/5.2/package.json:13]:
```suggestion
    "typescript": "~5.2.2"
```

**Review by aqrln — APPROVED** [review]

**Pull request merged**